stdlib/postgres/postgres.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"net"
+
+	"cloud.google.com/go/cloudsqlconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/stdlib"
+)
+
+func init() {
+	sql.Register("cloudsql-postgres", &pgDriver{})
```

Comment:
This is pgx specific, so I think the title should be pgx specific. We probably also want to include version since it's part of the package?

So maybe "cloud-sql-pgx-v4"? Or "cloudsql/pgx/v4"?

Reply:
Because `database/sql` provides a generic SQL interface, I think we don't want to expose the underlying library in the driver name.

For example:

With `lib/pq`, you register a driver with:

``` golang
import _ "github.com/lib/pq"
```

And use a driver with:

``` golang
db, err := sql.Open("postgres", connStr)
```

I would guess `pgx` used `pgx` as the driver name because `postgres` was taken.

For `go-sql-driver/mysql`, you register a driver with:

``` golang
import _ "github.com/go-sql-driver/mysql"
```

And use a driver with:

```
db, err := sql.Open("mysql", "user:password@/dbname")
```

For `go-mssqldb`, you register a driver with:

``` golang
import _ "github.com/denisenkom/go-mssqldb"
```

And use a driver with:

``` golang
db, err := sql.Open("sqlserver", connStr)
```

So I think what we want to do here is this:

Register a driver with:

``` golang
import _ "cloud.google.com/go/cloudsqlconn/postgres" 
```

And use a driver with:

``` golang
db, err := sql.Open("cloudsql-postgres", connStr)
```

Or possibly to match the import path:

``` golang
db, err := sql.Open("cloudsqlconn-postgres", connStr)
```

If we end up needing to create new drivers (e.g., `pgx/v4` is replaced by a newer, better version), we could always do this:

``` golang
db, err := sql.Open("cloudsqlconn-postgres/v2", connStr)
```